lib/vmdb/plugins.rb

```diff
@@ -1,78 +1,77 @@
+require 'singleton'
+
 module Vmdb
   class Plugins
     include Singleton
+    include Enumerable
```

Comment:
Personal opinion:  I would go without `Enumerable` here, or at least apply it to the class instead of the instance.

For those who didn't catch this, since it wasn't mentioned in the commit message of the second commit message, there are a bunch of changes like this:

```diff
-    Vmdb::Plugins.instance.vmdb_plugins.each do |plugin|
+    Vmdb::Plugins.instance.each do |plugin|
```

Which is enabled by doing `include Enumerable` call, but I think it would look better like this:


```diff
-    Vmdb::Plugins.instance.vmdb_plugins.each do |plugin|
+    Vmdb::Plugins.each do |plugin|
```

So not sure if this would work from the class level, or if you would just have to implement the methods in another way, but the call to `instance.each` just seems like a poor interface and isn't terribly intuitive to me.

Reply:
My guess is that all that would be required for this would be the following:

```diff
 module Vmdb
   class Plugins
     include Singleton

+    class << self do
+      include Enumerable
+
+      def each(&block)
+        instance.all.each(&block)
+      end
+    end
   end
 end
```

But I didn't test this myself.

Reply:
>  but I think it would look better like this:

100% agree, however there is a bunch of code that's already relying on the current `Singleton` implementation (which is where `.instance` comes from).  I've never liked `Singleton`'s way of doing `.instance` either, but I was concerned about changing that in this PR...I think I can change it in a follow-up.

Reply:
I thought about this now and the `.instance` (implying singular) followed by `.each` (implying plural) is definitely weird.  Maybe I'll put back the `.instance.all.each` I had originally. 🤔

Reply:
I took your advice and pushed a new change that "keeps" the  `Singleton` `.instance` (it's useful for testing and also handles multiple threads and whatnot), but at the class level, just forwards all methods to the instance.  I was going to use `Forwardable`, but you have to enumerate every method you want to delegate, and that seemed human error prone, particularly with the Enumerable there.

Reply:
I like this last option.  I agree, `instance.each` looks weird.